beacon-chain/p2p/discovery.go

```diff
@@ -47,6 +52,88 @@ type quicProtocol uint16
 // quicProtocol is the "quic" key, which holds the QUIC port of the node.
 func (quicProtocol) ENRKey() string { return "quic" }
 
+type listenerWrapper struct {
+	*sync.RWMutex
```

Comment:
I know it's done a lot of places in the Prysm codebase, but I find always awkward the mutex is exposed to the user, since it's only used internally.
Also, [uber-go](https://github.com/uber-go/guide/blob/master/style.md#zero-value-mutexes-are-valid) guidelines advice not to embed the mutex.